File: insfrastructure/entities/bank_account_entity.py
from datetime import datetime
from uuid import UUID, uuid4
from insfrastructure.entities.person_entity import PersonEntity
from insfrastructure.enums.account_types import AccountType


class BankAccountEntity:
    def __init__(
            self, 
            account_number: int, 
            account_holder: PersonEntity, 
            balance: float, 
            account_type: AccountType,
            creation_date:datetime = None,
            ):
        self.id = uuid4()
        self.account_number = account_number
        self.account_holder = account_holder
        self.balance = balance
        self.account_type = account_type
        self.creation_date = creation_date if creation_date is not None else datetime.now()

File: typescript-impl/src/infrastructure/entities/bank_account_entity.ts
import { v4 as uuidv4 } from 'uuid';
import { PersonEntity } from './person_entity';
import { AccountType } from '../enums/account_type';

export class BankAccountEntity {
    id: string;
    account_number: number;
    account_holder: PersonEntity;
    balance: number;
    account_type: AccountType;
    creation_date: Date;

    constructor(
        account_number: number,
        account_holder: PersonEntity,
        balance: number,
        account_type: AccountType,
        creation_date: Date | null = null,
    ) {
        this.id = uuidv4();
        this.account_number = account_number;
        this.account_holder = account_holder;
        this.balance = balance;
        this.account_type = account_type;
        this.creation_date = creation_date ? creation_date : new Date();
    }
}
